src/queue.ts

```diff
@@ -212,17 +253,65 @@ export const worker = function(routingKey: string, func: Worker) {
   workers[routingKey].push(func);
 };
 
+// == RPC CALL ==
+
+export const rpc = function(
+    routingKey: string,
+    data: {} | Buffer,
+    headers?: Headers,
+    ttl?: number): PromiseLike<any> {
+  if (replyTo) {
+    const _ttl = ttl || 60 * 1000;
+    const correlationId = uuidv4();
+    const options: amqp.ExchangePublishOptions = {
+      contentType: "application/json",
+      replyTo,
+      correlationId,
+      expiration: _ttl.toString(),
+      headers: headers ? headers : {}
+    };
+    return new Promise((resolve, reject) => {
+      const timeout = setTimeout(() => {
+        clearTimeout(timeout);
```

Comment:
Don't we have to clear the callback here as well?

Reply:
Yes, good catch!

Reply:
and I'm getting some tslint errors.

Reply:
Both issues fixed now!